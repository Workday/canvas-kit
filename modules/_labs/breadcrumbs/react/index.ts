import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
