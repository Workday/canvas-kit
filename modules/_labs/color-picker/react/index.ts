export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
