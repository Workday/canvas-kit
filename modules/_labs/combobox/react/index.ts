import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
