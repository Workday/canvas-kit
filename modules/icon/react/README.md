# Canvas Kit Icon

Components assisting the rendering of Canvas accent, applet, and system icons.

A Workday asset package must be used with these components. No icons are bundled with this package.

To browse a full list of icons, take a look at the
[Canvas Design System resources](https://design.workday.com/resources) page.

## Installation

```sh
yarn add @workday/canvas-kit-react
```

or

```sh
yarn add @workday/canvas-kit-react-icon
```

### Design Assets

Install one of the following icon packages:

```sh
@workday/canvas-accent-icons-web
@workday/canvas-applet-icons-web
@workday/canvas-system-icons-web
```

# Accent Icons

## Usage

Use with `@workday/canvas-accent-icons-web`.

```tsx
import { colors } from '@workday/canvas-kit-react-core'
import { AccentIcon } from '@workday/canvas-kit-react-icon'
import { shieldIcon } from '@workday/canvas-accent-icons-web'

<AccentIcon icon={shieldIcon} />
<AccentIcon icon={shieldIcon} color={colors.pomegranate500} />
<AccentIcon icon={shieldIcon} size={80} />
```

## Static Properties

> None

## Component Props

### Required

#### `icon: CanvasIcon`

> Icon to display from `@workday/canvas-accent-icons-web`.

### Optional

#### `color: string`

> Icon color from `@workday/canvas-colors-web`.

Default: `colors.blueberry500`

---

#### `size: number`

> Size of the icon.

Default: `56`

---

#### `transparent: boolean`

> Toggle for transparent accent icon background. If false, the background fill will be white.

Default: `false`

---

#### `iconRef: React.Ref<HTMLSpanElement>`

> Returns the ref to the rendered icon.

---

# Applet Icons

## Usage

Use with `@workday/canvas-applet-icons-web`.

```tsx
import { AppletIcon } from '@workday/canvas-kit-react-icon'
import { benefitsIcon } from '@workday/canvas-applet-icons-web'

<AppletIcon icon={benefitsIcon} />
<AppletIcon icon={benefitsIcon} color={AppletIcon.Colors.Pomegranate} />
<AppletIcon icon={benefitsIcon} size={60} />
```

## Static Properties

#### `Colors: BrandingColor`

> An enum of the various Canvas hues (`Pomegranate`, `Blueberry`, `Cinnamon`, etc.).

```tsx
<AppletIcon icon={benefitsIcon} color={AppletIcon.Colors.Pomegranate} />
```

- `Cinnamon`
- `Peach`
- `ChiliMango`
- `Cantaloupe`
- `SourLemon`
- `JuicyPear`
- `Kiwi`
- `GreenApple`
- `Watermelon`
- `Jewel`
- `Toothpaste`
- `Blueberry`
- `Plum`
- `BerrySmoothie`
- `Blackberry`
- `IslandPunch`
- `GrapeSoda`
- `Pomegranate`
- `FruitPunch`
- `RootBeer`
- `ToastedMarshmallow`
- `Cappuccino`
- `Licorice`
- `BlackPepper`

## Component Props

### Required

#### `icon: CanvasIcon`

> Icon to display from `@workday/canvas-applet-icons-web`.

### Optional

#### `color: AppletIcon.Colors`

> Icon color hue. Must use a member of the `AppletIcon.Colors` static enum.

Default: `AppletIcon.Colors.Blueberry`

---

#### `size: number`

> Size of the icon in `px`.

Default: `92`

---

#### `iconRef: React.Ref<HTMLSpanElement>`

> Returns the ref to the rendered icon.

---

# System Icons

## Usage

Use with `@workday/canvas-system-icons-web`.

```tsx
import { colors } from '@workday/canvas-kit-react-core'
import { SystemIcon } from '@workday/canvas-kit-react-icon'
import { activityStreamIcon } from '@workday/canvas-system-icons-web'

<SystemIcon icon={activityStreamIcon} />
<SystemIcon icon={activityStreamIcon} color={colors.blueberry500} />
<SystemIcon
  icon={activityStreamIcon}
  accent={colors.frenchVanilla100}
  fill={colors.blueberry500}
  background={colors.blueberry500}
/>
<SystemIcon icon={activityStreamIcon} size={48} />
```

## Static Properties

> None

## Component Props

### Required

#### `icon: CanvasIcon`

> Icon to display from `@workday/canvas-system-icons-web`

### Optional

#### `size: number`

> Size of the icon in `px`.

Default: `24`

---

#### `color: string`

> Icon color. This will define `accent` and `fill`. `accent` and `fill` will override this property
> if defined.

Default: `iconColors.standard`

---

#### `colorHover: string`

> Hover color of the icon. This will define `accentHover` and `fillHover`.

Default: `iconColors.hover`

---

#### `accent: string`

> `.wd-icon-accent` color. This will override `color`.

---

#### `accentHover: string`

> `.wd-icon-accent` color on hover. This will override `colorHover`.

---

#### `background: string`

> `.wd-icon-background` color.

Default: `'transparent'`

---

#### `backgroundHover: string`

> `.wd-icon-background` color on hover.

Default: `'transparent'`

---

#### `fill: string`

> `.wd-icon-fill` color. This will override `color`.

---

#### `fillHover: string`

> `.wd-icon-fill` color on hover. This will override `colorHover`.

---

#### `iconRef: React.Ref<HTMLSpanElement>`

> Returns the ref to the rendered icon.

---

# System Icon Circle

A system icon with a colored circular background. Icon color will be determined based on contrast
against the background color provided.

## Usage

Use with `@workday/canvas-accent-icons-web`.

```tsx
import { colors } from '@workday/canvas-kit-react-core'
import { SystemIconCircle } from '@workday/canvas-kit-react-icon'
import { shieldIcon } from '@workday/canvas-accent-icons-web'

<SystemIconCircle icon={shieldIcon} />
<SystemIconCircle icon={shieldIcon} background={colors.pomegranate500} />
```

## Static Properties

> None

## Component Props

### Required

#### `icon: CanvasIcon`

> Icon to display from `@workday/canvas-accent-icons-web`.

### Optional

#### `background: string`

> Background color from `@workday/canvas-colors-web`.

Default: `colors.soap300`

---

#### `size: SystemIconCircleSize | number`

> Size of the icon.

Default: `SystemIconCircleSize.l` (`40`)

---

#### `iconRef: React.Ref<HTMLSpanElement>`

> Returns the ref to the rendered icon.

---

# Graphics

## Usage

Use with `@workday/canvas-graphics-web`.

```tsx
import { colors } from '@workday/canvas-kit-react-core'
import { Graphic } from '@workday/canvas-kit-react-icon'
import { badgeAchievementGraphic } from '@workday/canvas-graphics-web'

<Graphic src={badgeAchievementGraphic} />
<Graphic src={badgeAchievementGraphic} width={80}/>
<Graphic src={badgeAchievementGraphic} height={80}/>
<Graphic src={badgeAchievementGraphic} grow={true} />
```

## Static Properties

#### `Size: SystemIconCircleSize`

> The size of the circle background. The icon will scale to fit.

| Size | Pixels |
| ---- | ------ |
| xs   | 16     |
| s    | 24     |
| m    | 32     |
| l    | 40     |
| xl   | 64     |
| xxl  | 120    |

## Component Props

### Required

#### `src: CanvasGraphic`

> Graphic to display from `@workday/canvas-graphics-web`

### Optional

#### `width: number | string`

> Graphic width in `px`. `width` takes precedence over `height` in order to preserve the graphic's
> ratio.

Default: `width of graphic`

---

#### `height: number | string`

> Graphic height in `px`. If set, `width` will be set to `100%`.

Default: `height of graphic`

---

#### `grow: boolean`

> Expand graphic to fit container. `grow` takes precedence over both `width` and `height`.

Default: `false`

#### `iconRef: React.Ref<HTMLSpanElement>`

> Returns the ref to the rendered icon.

---
