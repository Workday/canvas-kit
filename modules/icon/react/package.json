{
  "name": "@workday/canvas-kit-react-icon",
  "version": "3.5.0",
  "description": "Canvas icons",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "http://github.com/Workday/canvas-kit/tree/master/modules/icon/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts",
    "ts3.5/**/*"
  ],
  "typesVersions": {
    "<=3.5": {
      "*": [
        "ts3.5/*"
      ]
    }
  },
  "scripts": {
    "watch": "yarn build:es6 -w",
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build:downlevel-dts": "yarn run downlevel-dts dist ts3.5/dist",
    "build": "npm-run-all --parallel build:cjs build:es6 --sequential build:downlevel-dts",
    "depcheck": "node ../../../utils/check-dependencies-exist.js"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "icon"
  ],
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  },
  "dependencies": {
    "@emotion/core": "^10.0.28",
    "@emotion/is-prop-valid": "^0.8.2",
    "@emotion/styled": "^10.0.27",
    "@workday/canvas-kit-react-common": "^3.5.0",
    "@workday/canvas-kit-react-core": "^3.5.0",
    "@workday/design-assets-types": "^0.2.4"
  },
  "devDependencies": {
    "@workday/canvas-accent-icons-web": "^1.0.0",
    "@workday/canvas-applet-icons-web": "^0.17.50",
    "@workday/canvas-system-icons-web": "^1.0.20"
  }
}
