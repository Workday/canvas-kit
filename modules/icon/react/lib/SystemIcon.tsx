import * as React from 'react';
import {iconColors} from '@workday/canvas-kit-react-core';
import {CanvasSystemIcon, CanvasIconTypes} from '@workday/design-assets-types';
import {CSSObject} from 'create-emotion';
import Icon from './Icon';
import {SpanProps} from './types';

export interface SystemIconStyles {
  accent?: string;
  accentHover?: string;
  background?: string;
  backgroundHover?: string;
  color?: string;
  colorHover?: string;
  fill?: string;
  fillHover?: string;
}

export interface SystemIconProps extends SystemIconStyles {
  icon: CanvasSystemIcon;
  size?: number;
}

export const systemIconStyles = ({
  accent,
  accentHover,
  background = 'transparent',
  backgroundHover = 'transparent',
  color = iconColors.standard,
  colorHover = iconColors.hover,
  fill,
  fillHover,
}: SystemIconStyles): CSSObject => ({
  '& .wd-icon-fill': {
    fill: fill || color,
  },
  ':hover .wd-icon-fill': {
    fill: fillHover || colorHover,
  },
  '& .wd-icon-accent': {
    fill: accent || color,
  },
  ':hover .wd-icon-accent': {
    fill: accentHover || colorHover,
  },
  '& .wd-icon-background': {
    fill: background,
  },
  ':hover .wd-icon-background': {
    fill: backgroundHover,
  },
});

export default class SystemIcon extends React.Component<SpanProps & SystemIconProps> {
  render() {
    const {
      icon,
      accent,
      accentHover,
      background,
      backgroundHover,
      color,
      colorHover,
      fill,
      fillHover,
      size,
      ...elemProps
    } = this.props;

    const style = systemIconStyles({
      accent,
      accentHover,
      background,
      backgroundHover,
      color,
      colorHover,
      fill,
      fillHover,
    });

    return (
      <Icon
        src={icon}
        type={CanvasIconTypes.System}
        size={size}
        styles={style}
        elemProps={elemProps}
      />
    );
  }
}
