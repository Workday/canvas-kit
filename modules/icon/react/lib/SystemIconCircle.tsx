import * as React from 'react';
import styled from '@emotion/styled';
import isPropValid from '@emotion/is-prop-valid';
import {colors, borderRadius} from '@workday/canvas-kit-react-core';
import SystemIcon, {SystemIconProps} from './SystemIcon';
import {CanvasSystemIcon} from '@workday/design-assets-types';
import {pickForegroundColor} from '@workday/canvas-kit-react-common';

export enum SystemIconCircleSize {
  xs = 16,
  s = 24,
  m = 32,
  l = 40,
  xl = 64,
  xxl = 120,
}

export interface SystemIconCircleProps extends Pick<SystemIconProps, 'iconRef'> {
  /**
   * The background color of the SystemIconCircle from `@workday/canvas-colors-web`.
   * @default colors.soap300
   */
  background?: string;
  /**
   * The icon to display from `@workday/canvas-accent-icons-web`.
   */
  icon: CanvasSystemIcon;
  /**
   * The size of the SystemIconCircle.
   * @default SystemIconCircleSize.l
   */
  size?: SystemIconCircleSize | number;
}

const Container = styled('div', {
  shouldForwardProp: prop => isPropValid(prop) && prop !== 'size',
})<Pick<SystemIconCircleProps, 'background' | 'size'>>(
  {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 0,
    border: 0,
    borderRadius: borderRadius.circle,
    boxSizing: 'border-box',
    overflow: 'hidden',
    '& img': {
      width: '100%',
      height: '100%',
    },
  },
  ({background}) => ({
    background: background,
  }),
  ({size}) => ({
    width: size,
    height: size,
  })
);

export default class SystemIconCircle extends React.Component<SystemIconCircleProps> {
  static Size = SystemIconCircleSize;

  render() {
    const {
      background = colors.soap200,
      size = SystemIconCircleSize.l,
      icon,
      iconRef,
      ...elemProps
    } = this.props;

    const iconColor = pickForegroundColor(background, 'rgba(0,0,0,0.65)', colors.frenchVanilla100);
    const iconSize = size * 0.625;

    return (
      <Container background={background} size={size} {...elemProps}>
        <SystemIcon
          icon={icon}
          color={iconColor}
          colorHover={iconColor}
          size={iconSize}
          iconRef={iconRef}
        />
      </Container>
    );
  }
}
