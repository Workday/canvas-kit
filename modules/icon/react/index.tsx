import AccentIcon from './lib/AccentIcon';
import AppletIcon from './lib/AppletIcon';
import SystemIcon from './lib/SystemIcon';
import SystemIconCircle from './lib/SystemIconCircle';
import Graphic from './lib/Graphic';
import Svg from './lib/Svg';

export {AccentIcon, AppletIcon, SystemIcon, SystemIconCircle, Graphic, Svg};

export * from './lib/AccentIcon';
export * from './lib/AppletIcon';
export * from './lib/SystemIcon';
export * from './lib/SystemIconCircle';
export * from './lib/Graphic';
export * from './lib/Svg';
