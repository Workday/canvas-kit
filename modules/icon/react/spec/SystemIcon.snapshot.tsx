import * as renderer from 'react-test-renderer';
import SystemIcon from '../lib/SystemIcon';
import SystemIconCircle from '../lib/SystemIconCircle';
import * as React from 'react';
import {colors} from '@workday/canvas-kit-react-core';
import {activityStreamIcon} from '@workday/canvas-system-icons-web';

describe('System Icon Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<SystemIcon icon={activityStreamIcon} />);
    expect(component).toMatchSnapshot();
  });
});

describe('System Icon Circle Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<SystemIconCircle icon={activityStreamIcon} />);
    expect(component).toMatchSnapshot();
  });
  test('renders as expected with colored background', () => {
    const component = renderer.create(
      <SystemIconCircle icon={activityStreamIcon} background={colors.blueberry400} />
    );
    expect(component).toMatchSnapshot();
  });
});
