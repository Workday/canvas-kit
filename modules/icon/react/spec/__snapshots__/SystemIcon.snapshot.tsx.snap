// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`System Icon Circle Snapshots renders as expected 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  border: 0;
  border-radius: 100%;
  box-sizing: border-box;
  overflow: hidden;
  background: #e8ebed;
  width: 40px;
  height: 40px;
}

.emotion-1 img {
  width: 100%;
  height: 100%;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: rgba(0,0,0,0.65);
}

.emotion-0:hover .wd-icon-fill {
  fill: rgba(0,0,0,0.65);
}

.emotion-0 .wd-icon-accent {
  fill: rgba(0,0,0,0.65);
}

.emotion-0:hover .wd-icon-accent {
  fill: rgba(0,0,0,0.65);
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-0 svg {
  width: 25px;
  height: 25px;
}

<div
  className="emotion-1 emotion-2"
>
  <span
    className="emotion-0"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-activity-stream wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M20 17V4H4v13z\\" class=\\"wd-icon-background\\"/><path d=\\"M11 19l-3.602 2.701c-.22.165-.398.083-.398-.196V19H3.49a.496.496 0 0 1-.49-.496V3.496A.5.5 0 0 1 3.5 3h17a.5.5 0 0 1 .5.496v15.008a.503.503 0 0 1-.51.496H11zM5 5v12h14V5H5z\\" class=\\"wd-icon-fill\\"/><path d=\\"M7 11.505c0-.279.228-.505.491-.505h5.018a.49.49 0 0 1 .491.505v.99a.503.503 0 0 1-.491.505H7.491A.49.49 0 0 1 7 12.495v-.99zm0-3C7 8.226 7.228 8 7.51 8h8.98a.5.5 0 0 1 .51.505v.99a.507.507 0 0 1-.51.505H7.51A.5.5 0 0 1 7 9.495v-.99z\\" class=\\"wd-icon-accent\\"/></g></svg>",
      }
    }
  />
</div>
`;

exports[`System Icon Circle Snapshots renders as expected with colored background 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  border: 0;
  border-radius: 100%;
  box-sizing: border-box;
  overflow: hidden;
  background: #0875e1;
  width: 40px;
  height: 40px;
}

.emotion-1 img {
  width: 100%;
  height: 100%;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-fill {
  fill: #ffffff;
}

.emotion-0 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-accent {
  fill: #ffffff;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-0 svg {
  width: 25px;
  height: 25px;
}

<div
  className="emotion-1 emotion-2"
>
  <span
    className="emotion-0"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-activity-stream wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M20 17V4H4v13z\\" class=\\"wd-icon-background\\"/><path d=\\"M11 19l-3.602 2.701c-.22.165-.398.083-.398-.196V19H3.49a.496.496 0 0 1-.49-.496V3.496A.5.5 0 0 1 3.5 3h17a.5.5 0 0 1 .5.496v15.008a.503.503 0 0 1-.51.496H11zM5 5v12h14V5H5z\\" class=\\"wd-icon-fill\\"/><path d=\\"M7 11.505c0-.279.228-.505.491-.505h5.018a.49.49 0 0 1 .491.505v.99a.503.503 0 0 1-.491.505H7.491A.49.49 0 0 1 7 12.495v-.99zm0-3C7 8.226 7.228 8 7.51 8h8.98a.5.5 0 0 1 .51.505v.99a.507.507 0 0 1-.51.505H7.51A.5.5 0 0 1 7 9.495v-.99z\\" class=\\"wd-icon-accent\\"/></g></svg>",
      }
    }
  />
</div>
`;

exports[`System Icon Snapshots renders as expected 1`] = `
.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

<span
  className="emotion-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-activity-stream wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M20 17V4H4v13z\\" class=\\"wd-icon-background\\"/><path d=\\"M11 19l-3.602 2.701c-.22.165-.398.083-.398-.196V19H3.49a.496.496 0 0 1-.49-.496V3.496A.5.5 0 0 1 3.5 3h17a.5.5 0 0 1 .5.496v15.008a.503.503 0 0 1-.51.496H11zM5 5v12h14V5H5z\\" class=\\"wd-icon-fill\\"/><path d=\\"M7 11.505c0-.279.228-.505.491-.505h5.018a.49.49 0 0 1 .491.505v.99a.503.503 0 0 1-.491.505H7.491A.49.49 0 0 1 7 12.495v-.99zm0-3C7 8.226 7.228 8 7.51 8h8.98a.5.5 0 0 1 .51.505v.99a.507.507 0 0 1-.51.505H7.51A.5.5 0 0 1 7 9.495v-.99z\\" class=\\"wd-icon-accent\\"/></g></svg>",
    }
  }
/>
`;
